### PR TITLE
Add support in BasicHtmlWebResponseObject for open-ended input tags

### PR DESCRIPTION
**Pull request by farmerau**
# PR Summary

Per #16126, `BasicHtmlWebResponseObject` didn't fully support valid html for `input` fields. Specifically, for an `Invoke-WebRequest` that returns this valid html:
```html
<html>
    <body>
        <input name="foo" value="bar">
    </body>
</html>
```
would have a null `InputFields` value.

To solve this, I've added an alternative construct to the regular expression (as recommended by the author of the issue) that requires neither a closing slash '/' nor a closing input tag '</input>'.

Additionally, I've created Pester test cases to validate the outcome of the updated expression.
## PR Context

(For reference #15973)

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [x] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 1]
```diff
@@ -1933,6 +1933,59 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
...
+        It 'correctly parses an input tag with no end tag' {
+            $query = @{
+                contenttype = 'text/html'
+                body        = "<html><body><input name='foo' value='bar'></body></html>"
+            }
+            $uri = Get-WebListenerUrl -Test 'Response' -Query $query
+            $response = Invoke-WebRequest -Uri $uri -UseBasicParsing
+            $response.Error | Should -BeNullOrEmpty
+            $response.InputFields | Should -Not -BeNullOrEmpty
+            $response.InputFields.FindByName('foo') | Should -Not -BeNullOrEmpty
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:1946]: Can we check exact result?

Comment (farmerau) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:1946]: Sure, I'll address this when I wrap up with my day job.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 2]
```diff
@@ -219,7 +219,7 @@ private static void EnsureHtmlParser()
 
             if (s_inputFieldRegex == null)
             {
-                s_inputFieldRegex = new Regex(@"<input\s+[^>]*(/>|>.*?</input>)",
+                s_inputFieldRegex = new Regex(@"<input\s+[^>]*(/>|>|>.*?</input>)",
```
Comment (vexx32) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs:222]:
```suggestion
                s_inputFieldRegex = new Regex(@"<input\s+[^>]*(/?>|>.*?</input>)",
```

I think this is a bit of an easier to read / simpler regex expression that should have the same result.

**Review by iSazonov — APPROVED** [review]

**Comment by farmerau** [comment]
@iSazonov is there a next step for merging? I'm not an authorized member. From what I can tell, it should be okay to happen 1 business day after approval.

**Comment by iSazonov** [comment]
@farmerau We are waiting MSFT members to review. Since it is Regex they can find compliance issues.

/cc @daxian-dbw @TravisEz13 friendly ping.

**Review by daxian-dbw — APPROVED** [review]

**Comment by iSazonov** [comment]
@farmerau Thanks for your contribution!

**Pull request merged**